### PR TITLE
Update times in setup_*forcing methods

### PR DESCRIPTION
**Pull request by hboisgon**
see #97

### CODE REVIEW

**Review by DirkEilander — CHANGES_REQUESTED** [review]
suggestion to do check on times in write_forcing

File: hydromt_wflow/wflow.py [hunk 1]
```diff
@@ -1859,24 +1858,39 @@ def write_forcing(
                         self.write_config()  # re-write config
                         fn_out = fn_default_path
 
+            # Check if all dates between (starttime, endtime) are in all da forcing
+            start = pd.to_datetime(self.get_config("starttime"))
+            end = pd.to_datetime(self.get_config("endtime"))
+            missings = False
+            for da in self.forcing.values():
+                if "time" in da.coords:
```
Comment (DirkEilander) [inline hydromt_wflow/wflow.py:1866]: time should always be in forcing right? Why did you add this if statement?

File: hydromt_wflow/wflow.py [hunk 2]
```diff
@@ -1859,24 +1858,39 @@ def write_forcing(
...
+                    times = da.time.values
+                    if pd.to_datetime(start) < pd.to_datetime(times[0]):
```
Comment (DirkEilander) [inline hydromt_wflow/wflow.py:1868]: start is already a datetime object, you can skip the `pd.to_datetime(start)` here. And same for end in next if-statement. See also comment on L1875

File: hydromt_wflow/wflow.py [hunk 3]
```diff
@@ -1859,24 +1858,39 @@ def write_forcing(
...
+                        start = pd.to_datetime(times[0])
+                        missings = True
+                    if pd.to_datetime(end) > pd.to_datetime(times[-1]):
+                        end = pd.to_datetime(times[-1])
+                        missings = True
+                        start = pd.to_datetime(times[0]).to_pydatetime()
```
Comment (DirkEilander) [inline hydromt_wflow/wflow.py:1874]: I think this line should be deleted as it is in the wrong if statement

File: hydromt_wflow/wflow.py [hunk 4]
```diff
@@ -1859,24 +1858,39 @@ def write_forcing(
...
+                        end = pd.to_datetime(times[-1]).to_pydatetime()
```
Comment (DirkEilander) [inline hydromt_wflow/wflow.py:1875]: I suggest to remove the `to_pydatatime()` here to always keep `end` (and `start`) as datetime object and only do the conversion when you update the config on L1881

File: hydromt_wflow/wflow.py [hunk 5]
```diff
@@ -1859,24 +1858,39 @@ def write_forcing(
...
+                    f"Not all dates found in precip_fn changing starttime to {start} and endtime to {end} in the toml."
+                )
+                self.set_config("starttime", start.to_pydatetime())
+                self.set_config("endtime", end.to_pydatetime())
+                self.write_config()
+
             # merge, process and write forcing
             ds = xr.merge([da.reset_coords(drop=True) for da in self.forcing.values()])
+            ds.raster.set_crs(self.crs)
+            ds = ds.sel({"time": slice(start, end)})
```
Comment (DirkEilander) [inline hydromt_wflow/wflow.py:1888]: I suggest to move this to the `if missings:` statement.

**Review by DirkEilander — CHANGES_REQUESTED** [review]
Good improvement @hboisgon! 
One remark: I think L1874 should be removed / is misplaced. 
Otherwise some small suggestions.

File: hydromt_wflow/wflow.py (hunk 1 above)
Comment (hboisgon) [reply hydromt_wflow/wflow.py:1866]: Apparently spatial_ref ends up as a da in self.forcing and it does not have time dimension

**Review by DirkEilander — APPROVED** [review]
